ingest/importer/submission/submission.py

```diff
@@ -0,0 +1,90 @@
+import logging
+from typing import List
+
+from ingest.api.ingestapi import IngestApi
+from ingest.importer.submission.entity import Entity
+from ingest.importer.submission.entity_map import EntityMap
+
+format = '[%(filename)s:%(lineno)s - %(funcName)20s() ] %(asctime)s - %(name)s - %(levelname)s - %(message)s'
+logging.basicConfig(format=format)
+
+ENTITY_LINK = {
+    'biomaterial': 'biomaterials',
+    'process': 'processes',
+    'file': 'files',
+    'protocol': 'protocols',
+    'project': 'projects',
+    'submission_envelope': 'submissionEnvelopes'
+}
+
+
+class Submission(object):
+    def __init__(self, ingest_api: IngestApi, submission_url: str):
+        self.ingest_api = ingest_api
+        self.submission_url = submission_url
+        self.metadata_dict = {}
+        self.manifest = None
+        self.logger = logging.getLogger(__name__)
+
+    def is_update(self):
+        submission = self.ingest_api.get_submission(self.submission_url)
+        return submission.get('isUpdate', False)
+
+    def get_submission_url(self):
+        return self.submission_url
+
+    def add_entity(self, entity: Entity):
+        link_name = ENTITY_LINK[entity.type]
+
+        # TODO: how to get filename?!!!
```

Comment:
Does this TODO needs to be done? If so, do it. If not - delete.

Reply:
I don't think there's any way to get the filename, so I just deleted it.